app/src/main/java/com/vultisig/wallet/presenter/keysign/KeysignViewModel.kt

```diff
@@ -228,30 +229,39 @@ internal class KeysignViewModel(
         }
     }
 
-    private fun getSignedTransaction(): SignedTransactionResult {
+    private suspend fun getSignedTransaction(): SignedTransactionResult {
         val swapPayload = keysignPayload.swapPayload
+
+        var nonceAcc = BigInteger.ZERO
+
+        if (keysignPayload.approvePayload != null) {
+            val signedTransaction = THORChainSwaps(vault.pubKeyECDSA, vault.hexChainCode)
+                .getSignedApproveTransaction(
+                    keysignPayload.approvePayload,
+                    keysignPayload,
+                    signatures
+                )
+
+            val evmApi = evmApiFactory.createEvmApi(keysignPayload.coin.chain)
+            evmApi.sendTransaction(signedTransaction.rawTransaction)
```

Comment:
Here , we might want to display both the approval transaction hash and send transaction hash in future , probable will need to return it , so we can display it on UI part